### PR TITLE
Timing data in Sampler

### PR DESCRIPTION
**Pull request by beckykd**
Should be ready by IEEE

### CODE REVIEW

File: docs/guides/monitor-job.mdx [hunk 1]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
 In Qiskit, use `job.cancel()` to cancel a job.
 
+<span id="execution-spans"></span>
+## View Sampler execution spans
+
+The results of `SamplerV2` jobs executed in Qiskit Runtime contain execution timing information in their metadata.
+This timing information can be used to place upper and lower timestamp bounds on when particular shots were executed on the QPU.
+Shots are grouped into `~.ExecutionSpan`s, each of which indicates a start time, a stop time, and a specification of which shots were collected in the span.
+
+An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a boolean mask that is ``True`` for all shots executing during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2,3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:60]:
```suggestion
An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a boolean mask that is ``True`` for all shots executing during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2, 3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```

File: docs/guides/monitor-job.mdx [hunk 2]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
 
 In Qiskit, use `job.cancel()` to cancel a job.
 
+<span id="execution-spans"></span>
+## View Sampler execution spans
+
+The results of `SamplerV2` jobs executed in Qiskit Runtime contain execution timing information in their metadata.
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:56]: Out of curiosity, is there some kind of magic that goes through these markdowns and convert class names to API links?

**Review by ihincks — COMMENTED** [review]
Thanks Becky!

File: docs/guides/monitor-job.mdx (hunk 2 above)
Comment (beckykd) [reply docs/guides/monitor-job.mdx:56]: No; I need to add links if we want them.

Comment (beckykd) [reply docs/guides/monitor-job.mdx:56]: I added a link to the first reference.

File: docs/guides/monitor-job.mdx [hunk 3]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
 In Qiskit, use `job.cancel()` to cancel a job.
 
+<span id="execution-spans"></span>
+## View Sampler execution spans
+
+The results of [`SamplerV2`](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.SamplerV2) jobs executed in Qiskit Runtime contain execution timing information in their metadata.
+This timing information can be used to place upper and lower timestamp bounds on when particular shots were executed on the QPU.
+Shots are grouped into `~.ExecutionSpan`s, each of which indicates a start time, a stop time, and a specification of which shots were collected in the span.
+
+An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a boolean mask that is ``True`` for all shots executing during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2, 3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:60]:
```suggestion
An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a boolean mask that is ``True`` for all shots executed during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2, 3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```

File: docs/guides/monitor-job.mdx [hunk 4]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
+```python
+# Get the mask of the 1st PUB for the 0th span.
+mask = spans[0].mask(1)
+
+# Decide whether the 0th shot of parameter set (1, 2) occurred in this span.
+in_this_span = mask[1, 2, 0]
+
+# Create a new bit array containing only the PUB-1 data collected during this span.
+bits = pub_results[1].data.meas
+filtered_data = BitArray(bits.array(mask), bits.num_bits)
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:71]:
```suggestion
filtered_data = BitArray(bits.array[mask], bits.num_bits)
```

File: docs/guides/monitor-job.mdx [hunk 5]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
+```
+
+To view execution span information, review the metadata of the result returned by `SamplerV2`, which comes in the form of an `~ExecutionSpans` object. This object is a list-like container containing subclass instances of `ExecutionSpan`s such as `SliceSpan`:
+
+```python
+pub_results = job.result()
+spans = pub_results.metadata["execution"]["execution_spans"]
+for span in spans:
+     print(span)
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:80]: Is there any chance we could show an example of what the output looks like? Here,

```bash
SliceSpan(<start='2024-09-09 11:48:21', stop='2024-09-09 11:48:22', size=1280>)
SliceSpan(<start='2024-09-09 11:48:21', stop='2024-09-09 11:48:25', size=1280>)
SliceSpan(<start='2024-09-09 11:48:21', stop='2024-09-09 11:48:24', size=1280>)
SliceSpan(<start='2024-09-09 11:48:21', stop='2024-09-09 11:48:27', size=1280>)
SliceSpan(<start='2024-09-09 11:48:21', stop='2024-09-09 11:48:26', size=1280>)
...
```

File: docs/guides/monitor-job.mdx [hunk 6]
```diff
@@ -50,6 +50,64 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
+Extract and inspect a particular span:
+
+```python
+spans.sort()
+print(" Start of first span:", spans[0].start)
+print("   End of first span:", spans[0].stop)
+print("#shots in first span:", spans[0].size)
+```
+
+<Admonition>
```
Comment (ihincks) [inline docs/guides/monitor-job.mdx:108]: This is more about the theme than anything, but Admonition seems like rather a subtle change over the regular paragraph style, and not very admonishing :P

image

Comment (beckykd) [reply docs/guides/monitor-job.mdx:108]: Oh, I didn't put a type.  That will help!

File: docs/guides/monitor-job.mdx [hunk 7]
```diff
@@ -105,7 +115,7 @@ print("   End of first span:", spans[0].stop)
 print("#shots in first span:", spans[0].size)
 ```
 
-<Admonition>
+<Admonition type ="note">
```
Comment (Eric-Arellano) [inline docs/guides/monitor-job.mdx:118]:
```suggestion
<Admonition type="note">
```

File: docs/guides/monitor-job.mdx [hunk 8]
```diff
@@ -18,7 +18,7 @@ Use the job instance to check the job status or retrieve the results by calling
...
 | ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
 | job.result()                  | Review job results immediately after the job completes. Job results are available after the job completes. Therefore, job.result() is a blocking call until the job completes.                               |
-| job.job\_id()                 | Return the ID that uniquely identifies that job. Retrieving the job results at a later time requires the job ID. Therefore, it is recommended that you save the IDs of jobs you might want to retrieve later. |
+| job.job\_id()                 | Return the ID that uniquely identifieshttps://ibm-quantumcomputing.slack.com/archives/CSV6D7KKL/p1725881005460219 that job. Retrieving the job results at a later time requires the job ID. Therefore, it is recommended that you save the IDs of jobs you might want to retrieve later. |
```
Comment (abbycross) [inline docs/guides/monitor-job.mdx:21]:
```suggestion
| job.job\_id()                 | Return the ID that uniquely identifies that job. Retrieving the job results at a later time requires the job ID. Therefore, it is recommended that you save the IDs of jobs you might want to retrieve later. |
```

File: docs/guides/monitor-job.mdx [hunk 9]
```diff
@@ -50,6 +50,74 @@ You can cancel a job from the IBM Quantum Platform dashboard either on the Jobs
...
 In Qiskit, use `job.cancel()` to cancel a job.
 
+<span id="execution-spans"></span>
+## View Sampler execution spans
+
+The results of [`SamplerV2`](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.SamplerV2) jobs executed in Qiskit Runtime contain execution timing information in their metadata.
+This timing information can be used to place upper and lower timestamp bounds on when particular shots were executed on the QPU.
+Shots are grouped into `~.ExecutionSpan`s, each of which indicates a start time, a stop time, and a specification of which shots were collected in the span.
+
+An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a boolean mask that is ``True`` for all shots executed during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2, 3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```
Comment (abbycross) [inline docs/guides/monitor-job.mdx:60]:
```suggestion
An execution span specifies which data was executed during its window by providing an `~.ExecutionSpan.mask` method. This method, given any PUB index, returns a Boolean mask that is ``True`` for all shots executed during its window. PUBs are indexed by the order in which they were given to the Sampler run call. If, for example, a PUB has shape `(2, 3)` and was run with four shots, then the mask's shape is `(2, 3, 4)`.
```

File: docs/guides/primitive-input-output.ipynb [hunk 10]
```diff
@@ -466,6 +466,14 @@
     "for key, val in result[0].metadata.items():\n",
     "    print(f\"'{key}' : {val},\")"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12c8acfd-d189-45b7-a372-10c48bb13687",
+   "metadata": {},
+   "source": [
+    "For Sampler jobs, you can also review the result metadata to understand when certain data was run; called the [_execution span._](monitor-job#execution-spans)"
```
Comment (abbycross) [inline docs/guides/primitive-input-output.ipynb:475]:
```suggestion
    "For Sampler jobs, you can also review the result metadata to understand when certain data was run; this is called the [_execution span._](monitor-job#execution-spans)"
```

**Review by abbycross — APPROVED** [review]
A few tweaks then good to go!

**Pull request merged**